contrib/win32/win32compat/w32-sshfileperm.c

```diff
@@ -247,7 +249,50 @@ check_secure_folder_permission(const wchar_t* path_utf16, int read_ok)
 			continue;
 		}
 		else {
-			ret = -1;
+			log_on_stderr = 0;
+
+			PSID adminSid = NULL;
+			WCHAR adminName[UNLEN + 1];
+			WCHAR adminDomain[DNLEN + 1];
+			DWORD adminNameSize = UNLEN + 1;
+			DWORD adminDomainSize = DNLEN + 1;
+			DWORD adminSidSize = SECURITY_MAX_SID_SIZE;
+			PSID systemSid = NULL;
+			WCHAR systemName[UNLEN + 1];
+			WCHAR systemDomain[DNLEN + 1];
+			DWORD systemNameSize = UNLEN + 1;
+			DWORD systemDomainSize = DNLEN + 1;
+			DWORD systemSidSize = SECURITY_MAX_SID_SIZE;
+			SID_NAME_USE sidType;
+
+			adminSid = (PSID)malloc(SECURITY_MAX_SID_SIZE);
+			if (adminSid != NULL) {
+				if (CreateWellKnownSid(WinBuiltinAdministratorsSid, NULL, adminSid, &adminSidSize) != 0) {
+					if (LookupAccountSidW(NULL, adminSid, adminName, &adminNameSize, adminDomain, &adminDomainSize, &sidType) != 0) {
+						systemSid = (PSID)malloc(SECURITY_MAX_SID_SIZE);
+						if (systemSid != NULL) {
+							if (CreateWellKnownSid(WinLocalSystemSid, NULL, systemSid, &systemSidSize) != 0) {
+								if (LookupAccountSidW(NULL, systemSid, systemName, &systemNameSize, systemDomain, &systemDomainSize, &sidType) != 0) {
+									logit("Suggest restricting write permissions on '%S' folder to %S\\%S and %S\\%S.", path_utf16, systemDomain, systemName, adminDomain, adminName);
```

Comment:
The message would be more useful if instead of suggesting what it should be, indicate what makes it open